### PR TITLE
Allow MAVEN_HOME or M2_HOME for integration tests

### PR DESCRIPTION
**Pull request by cab105**
The IntegrationTestBaseClass is looking for the `MAVEN_HOME` environment variable, which only applies for Maven 1.x instead of `M2_HOME` which applies to Maven 2.x. This PR adds support for the second environment variable.

### CODE REVIEW

**Comment by cab105** [comment]
This PR includes an additional fix to ensure the version of spring-boot-starter-parent exists in Maven Central (3.x doesn't according to https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-parent)

File: applications/spring-shell/src/test/java/org/springframework/sbm/IntegrationTestBaseClass.java [hunk 1]
```diff
@@ -110,11 +110,18 @@ public abstract class IntegrationTestBaseClass {
 
     @BeforeAll
     public static void beforeAll() {
-        if (System.getenv("MAVEN_HOME") == null) {
+        String mvnHome = System.getenv("MAVEN_HOME");
+
+        if (mvnHome == null) {
+            mvnHome = System.getenv("M2_HOME");
```
Comment (fabapp2) [inline applications/spring-shell/src/test/java/org/springframework/sbm/IntegrationTestBaseClass.java:116]: 👍

File: components/sbm-openrewrite/src/test/java/org/springframework/sbm/openrewrite/maven/BumpParentVersionTest.java [hunk 2]
```diff
@@ -61,7 +61,8 @@ void bumpedParentPomVersionIsNotReflectedInModelTest() {
 
         List<Xml.Document> parsedPom = MavenParser.builder().build().parse(pomXml);
 
-        List<Result> results = new UpgradeParentVersion("org.springframework.boot", "spring-boot-starter-parent", "3.0.0-M3", null).run(parsedPom);
+        List<Result> results = new UpgradeParentVersion("org.springframework.boot", "spring-boot-starter-parent", "2.7.1", null).run(parsedPom);
```
Comment (fabapp2) [inline components/sbm-openrewrite/src/test/java/org/springframework/sbm/openrewrite/maven/BumpParentVersionTest.java:64]: The test was correct. Now it's setting the version of the parent pom to the version it already has and the `UpgradeParentPom` recipe will do nothing.
The issue (before upgrading OpenRewrite to a more recent version as on `main` now) was that the change to `3.0.0-M3` was applied in the `pom.xml` source (given that the milestone repo was configured)  but not reflected in the pom's model. 
This is the case now in `main` and the test proved that in latest OR version the change is reflected in the model too.
The comment should be removed as the call to `RefreshPomModel` in `OpenRewriteMavenBuildFile.upgradeParentVersion()` has been removed.

**Review by fabapp2 — CHANGES_REQUESTED** [review]

File: components/sbm-openrewrite/src/test/java/org/springframework/sbm/openrewrite/maven/BumpParentVersionTest.java (hunk 2 above)
Comment (fabapp2) [reply components/sbm-openrewrite/src/test/java/org/springframework/sbm/openrewrite/maven/BumpParentVersionTest.java:64]: Hm. Sounds like the root problem was that Rewrite couldn't retrieve the parent in `3.0.0-M3` from the configured milestone repo?
Actually, the test could be deleted. I'd say. 😬 
After removing the call to `RefreshPomModel` in `OpenRewriteMavenBuildFile.upgradeParentVersion()` after upgrading rewrite to  `7.26.0` changes to the parent version are reflected in the underlying model by rewrite automatically. The intention of the test was to remind us about this fact.

**Review by fabapp2 — CHANGES_REQUESTED** [review]

**Comment by cab105** [comment]
That's my take. Okay let me take it out and verify.

**Review by fabapp2 — APPROVED** [review]

**Pull request merged**